android/app/src/main/java/poke/rogue/helper/presentation/util/SpannableStringExtensions.kt

```diff
@@ -0,0 +1,63 @@
+package poke.rogue.helper.presentation.util
+
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.style.ForegroundColorSpan
+import android.text.style.ImageSpan
+import android.text.style.StyleSpan
+
+inline fun SpannableString.applySpans(configure: SpannableString.() -> Unit): SpannableString {
+    this.configure()
+    return this
+}
+
+fun SpannableString.drawable(
+    fullText: String,
+    iconDrawable: Drawable?,
+    iconSize: Int,
+) {
+    val iconIndex = fullText.indexOf("|")
+    if (iconIndex != -1) {
+        val drawable =
+            iconDrawable?.apply {
+                setBounds(0, 0, iconSize, iconSize)
+            }
+        val imageSpan = ImageSpan(drawable!!, ImageSpan.ALIGN_BOTTOM)
+        setSpan(imageSpan, iconIndex, iconIndex + 1, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+    }
+}
+
+fun SpannableString.color(
+    targetWord: String,
+    color: Int,
+    fullText: String,
+): SpannableString {
+    val startIndex = fullText.indexOf(targetWord)
+    if (startIndex != -1) {
+        this.setSpan(
+            ForegroundColorSpan(color),
+            startIndex,
+            startIndex + targetWord.length,
+            Spannable.SPAN_EXCLUSIVE_EXCLUSIVE,
+        )
+    }
+    return this
+}
+
+fun SpannableString.style(
```

Comment:
```suggestion
fun SpannableString.bold(
```

`bold` 처리만 해주는 것 같아서 `bold` 어떤가요?    
사실 비슷한 함수를 androidX 에서 제공해주고 있긴합니다. (위 color도 비슷하게!)

![image](https://github.com/user-attachments/assets/6cee719d-512d-4682-ab81-0224f9eb9b58)

Reply:
아예 styleSpan 을 파라미터로 받게 하겠습니다!